### DISCUSSION TITLE
Groupid for the OSGi Technology project

### DISCUSSION THREAD

**Discussion by maho7791** [osgi/osgi#599] [Polls]
We need to decide, which one to take.

**Comment by bosschaert** [comment]
The `org.osgi.` prefix was traditionally limited to spec APIs

**Comment by timothyjward** [comment]
This vote will remain open until the next OSGi specification call on July 19th.

**Comment by grgrzybek** [comment]
Here's my analysis...

If we browse to https://projects.eclipse.org/projects/technology and look at "RELATED PROJECTS" frame we can see kind of hierarchy. Most of the projects are in flat list, but there are two exceptions:
```
Eclipse Nebula - Supplemental Widgets for SWT
    Eclipse Nebula Incubator
    Eclipse Nebula NatTable
...
Eclipse Packaging Project
    Eclipse Marketplace Client™
```

related hrefs use another dot as subprojects:
```
/projects/technology.nebula
/projects/technology.nebula.incubator
/projects/technology.nebula.nattable
...
/projects/technology.packaging
/projects/technology.packaging.mpc
```

`/projects/technology.osgi-technology` is the _only_ project that uses "technology" in its name which is first (IMO) potential source of confusion, because there's no `org.eclipse.technology` groupId at https://repo1.maven.org/maven2/org/eclipse/

**That's why I think option 4 has least sense: `org.eclipse.technology.osgi`**

When comparing projects at https://projects.eclipse.org/projects/technology and subdirectories of https://repo1.maven.org/maven2/org/eclipse/:
* only 21 projects have clear 1:1 mapping: birt, collections, edc, jemo, jgit, jnosql, lsp4j, lyo, microprofile, nebula, osgi, packager, pass, passage, rdf4j, scout, sisu, steady, transformer, tycho, xpanse
* 73 projects from https://projects.eclipse.org/projects/technology have no obvious subdirectory at https://repo1.maven.org/maven2/org/eclipse/
* 104 subdirectories at https://repo1.maven.org/maven2/org/eclipse/ have no obvious matching project at https://projects.eclipse.org/projects/technology
* "osgi" project's "osgi" Maven Central subdirectory of org.eclipse is (as we know) real Maven _container_ for [OSGi® Specification Project](https://projects.eclipse.org/projects/technology.osgi)

https://projects.eclipse.org/projects/technology.osgi/downloads (and /developer) points to:
* https://repo.maven.apache.org/maven2/org/osgi/
* https://oss.sonatype.org/content/repositories/snapshots/org/osgi/
* https://github.com/osgi/osgi.git
(which is good)

https://projects.eclipse.org/projects/technology.osgi-technology/downloads still points to:
* https://repo.maven.apache.org/maven2/org/osgi/
* https://oss.sonatype.org/content/repositories/snapshots/org/osgi/

https://projects.eclipse.org/projects/technology.osgi-technology/developer points to:
* https://github.com/osgi/osgi-test.git
* https://github.com/osgi/slf4j-osgi.git
* https://github.com/osgi/osgi.enroute.git
* https://github.com/osgi/osgi.enroute.site.git
* https://github.com/osgi/jakartarest-osgi.git

which _could_ suggest two ways to go (principle of least surprise):
* either the GroupID doesn't contain `eclipse`, or
* these 5 projects are moved from github/osgi to github/eclipse

**So at this stage of analysis I'd vote for option 3: `org.osgi.technology`**

Let's think about option 3:
* https://repo1.maven.org/maven2/org/osgi/technology could be the parent group
* maybe existing `org.osgi.test.cases.xxx` could go (in newer versions) under https://repo1.maven.org/maven2/org/osgi/tck?
* there are already https://repo1.maven.org/maven2/org/osgi/osgi.enroute.xxx/ projects and https://repo1.maven.org/maven2/org/osgi/enroute _container_ - another source of confusion

Now some arguments for https://repo1.maven.org/maven2/org/eclipse/osgi/technology (no such option in the poll):
* https://repo1.maven.org/maven2/org/eclipse/osgi/ seems to be abandoned by Equinox since 2013
* we'd avoid `-` in group ID (which is not very common under org/eclipse anyway

At this stage (I'm not voting yet, waiting for your feedback ;), my order of preference is:
1. org.osgi.technology - because of what https://projects.eclipse.org/projects/technology.osgi/downloads parent project of "OSGi Technology" refers to
2. org.eclipse.osgi.technology (no option in the poll) - because `org.eclipse.osgi` is abandoned by Equinox
3. org.eclipse.osgi-technology (only because it matches https://projects.eclipse.org/projects/technology.osgi-technology)
4. org.eclipse.osgitech (because it sounds artificial)
5. org.eclipse.technology.osgi (because it'd suggest that EVERYTHING mentioned at https://projects.eclipse.org/projects/technology should go under `org.eclipse.technology` at some point

Thanks for your attention ;)

**Comment by maho7791** [comment]
The vote for a groupId is closed now!

The voters decided to go for: **org.eclipse.technology.osgi**

We will initiate the registration for this groupId at the Eclipse team as soon as the HIPP instance is available.
